src/main/java/com/atlassian/bitbucket/jenkins/internal/scm/BitbucketSCMSource.java

```diff
@@ -462,6 +478,16 @@ private void initialize() {
         }
     }
 
+    private BitbucketTag fetchBitbucketTag(BitbucketTagSCMHead head, TaskListener listener) {
+        BitbucketSCMSource.DescriptorImpl descriptor = (BitbucketSCMSource.DescriptorImpl) getDescriptor();
+        Optional<BitbucketServerConfiguration> mayBeServerConf = descriptor.getConfiguration(getServerId());
+        BitbucketServerConfiguration serverConfiguration = mayBeServerConf.get();
```

Comment:
This is a blind get. which has me a bit nervous. A quick and dirty fix would be to change the return type to be `Optional<BitbucketTag>` and then change this method to

```java
BitbucketSCMSource.DescriptorImpl descriptor = (BitbucketSCMSource.DescriptorImpl) getDescriptor();
        return descriptor.getConfiguration(getServerId()).map(serverConfiguration -> {BitbucketScmHelper scmHelper = descriptor.getBitbucketScmHelper(serverConfiguration.getBaseUrl(), getCredentials().orElse(null));
            return scmHelper.getTagClient(getProjectKey(), getRepositorySlug(), listener)
                    .getRemoteTag(head.getName());});
```